### DISCUSSION TITLE
Strange username in private repositories

### DISCUSSION THREAD

**Discussion by oxgl** [community/community#22517] [New to GitHub]
<p>Hello experts,<br>
I’ve created projects in VSCode, and uploaded to GitHub into my private repositories. Today I’ve checked my repositories on GitHub page, and I realized that GH shows username of a foreign user as creator - I don’t know who this user is:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/7/17e22a8565cb3c76dd18219d3aa19df76287a399.png" data-download-href="/uploads/short-url/3phxvuuelg84rJUEpKTguhp1G09.png?dl=1" title="3Zr93cW" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/35221549/181071577-0dbc813c-c3f6-4607-9dfe-171ffb2aa570.png" alt="3Zr93cW" data-base62-sha1="3phxvuuelg84rJUEpKTguhp1G09" width="690" height="163" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/17e22a8565cb3c76dd18219d3aa19df76287a399_2_690x163.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/7/17e22a8565cb3c76dd18219d3aa19df76287a399_2_1035x244.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/7/17e22a8565cb3c76dd18219d3aa19df76287a399.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/7/17e22a8565cb3c76dd18219d3aa19df76287a399_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">3Zr93cW</span><span class="informations">1224×290 22.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>But actually these uploads were made by me. Is this a bug in GH? I’ve also checked the username in terminal git config --list, and it shows the right username/e-mail address…<br>
How is this possible?<br>
Thank you!</p>
<p>BR,<br>
oxgl</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Check the mail address in the actual commits (using <code>git log</code>). Is that correct, too? Because the wrong user is shown I assume not. :wink:</p>

**Comment by oxgl** [comment]
<p>Hi!<br>
OMG! You are right! My fault. I’ve already checked the e-mail in the logs, and git settings. Today morning I checked again. I’m using privacy e-mail address <a href="mailto:id+username@users.noreply.github.com">id+username@users.noreply.github.com</a>. I saw this address in the settings, so I thought it’s OK. But when I copied the e-mail address from GitHub, I probably didn’t select the first digit in the ID:<br>
Address on GitHub starts with 3522…, but in my settings starts with 522…:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/f/9f97a2b5d956554d8e585ed7d11c843fb4b3d6a6.png" data-download-href="/uploads/short-url/mLONk5zfj1yvlooXKJaNmISf4d8.png?dl=1" title="github_account_vs_log" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/35221549/181071579-df660741-b204-425a-9c6b-6c02fa330d86.png" alt="github_account_vs_log" data-base62-sha1="mLONk5zfj1yvlooXKJaNmISf4d8" width="690" height="175" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/f/9f97a2b5d956554d8e585ed7d11c843fb4b3d6a6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">github_account_vs_log</span><span class="informations">764×194 28.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>But it’s still strange. Why is that user displayed instead of something like invalid or unknown user? How is it possible that for a 6 digit ID + my username GitHub shows a totally different user?</p>
<p>Anyway, thank you for your help!</p>
<p>oxgl</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="oxgl" data-post="3" data-topic="229238">
<div class="title">
<div class="quote-controls"></div>
 oxgl:</div>
<blockquote>
<p>How is it possible that for a 6 digit ID + my username GitHub shows a totally different user?</p>
</blockquote>
</aside>
<p>Based on the docs: The ID is the only thing that’s compared there, because that format was introduced so commits would still be recognized as yours if you change your username. So I assume the name part is actually ignored there. :sweat_smile:</p>

**Comment by oxgl** [comment]
<p>I see… :slight_smile: Thanks for the explanation and for your help! I really appreciate it!</p>
